pandas/core/groupby/groupbyindexing.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import annotations
+
+import numpy as np
+
+from pandas.util._decorators import doc
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .iloc to GroupBy.
+    """
+
+    @property
+    def iloc(self) -> _ilocGroupByIndexer:
```

Comment:
why would this take 2 arguments at all? that is needlessley complex; stick to row indexing.

Reply:
- To keep the same signature as df.iloc.
- Because it is trivial to implement. I am calling df.iloc internally anyway, so I just pass in the optional second index.
- To allow multiple filter operations in one call to avoid the performance overhead of chaining.